--- a/fs/reiserfs/super.c	2017-07-20 21:59:24.000000000 -0700
+++ b/fs/reiserfs/super.c	2017-07-21 16:03:01.809194381 -0700
@@ -108,7 +108,12 @@
 		sbi->work_queued = 0;
 	spin_unlock(&sbi->old_work_lock);
 
-	reiserfs_sync_fs(s, 1);
+        // lime-tech: avoid needless spin-ups
+	//reiserfs_sync_fs(s, 1);
+        reiserfs_write_lock(s);
+        reiserfs_flush_old_commits(s);
+        reiserfs_write_unlock(s);
+        //
 	up_read(&s->s_umount);
 }
 
