--- a/arch/x86/kvm/x86.c	2017-01-20 03:56:02.000000000 -0600
+++ b/arch/x86/kvm/x86.c	2017-01-25 14:45:39.746927978 -0600
@@ -109,7 +109,7 @@
 static bool __read_mostly ignore_msrs = 0;
 module_param(ignore_msrs, bool, S_IRUGO | S_IWUSR);

-static bool __read_mostly report_ignored_msrs = true;
+static bool __read_mostly report_ignored_msrs = false;
 module_param(report_ignored_msrs, bool, S_IRUGO | S_IWUSR);

 unsigned int min_timer_period_us = 500;
